v2/pkg/engine/datasource/pubsub_datasource/pubsub_datasource.go

```diff
@@ -81,81 +102,22 @@ func (p *Planner[T]) EnterField(ref int) {
 		return
 	}
 
-	topic := eventConfig.Topic
-	rex, err := regexp.Compile(`{{ args.([a-zA-Z0-9_]+) }}`)
-	if err != nil {
-		return
-	}
-	matches := rex.FindAllStringSubmatch(topic, -1)
-	if len(matches) != 1 || len(matches[0]) != 2 {
-		return
-	}
-	argName := matches[0][1]
-	// We need to find the argument in the operation
-	arg, ok := p.visitor.Operation.FieldArgument(ref, []byte(argName))
-	if !ok {
-		return
-	}
-	argValue := p.visitor.Operation.ArgumentValue(arg)
-	if argValue.Kind != ast.ValueKindVariable {
-		return
-	}
-	variableName := p.visitor.Operation.VariableValueNameBytes(argValue.Ref)
-	variableDefinition, ok := p.visitor.Operation.VariableDefinitionByNameAndOperation(p.visitor.Walker.Ancestors[0].Ref, variableName)
-	if !ok {
-		return
-	}
-	variableTypeRef := p.visitor.Operation.VariableDefinitions[variableDefinition].Type
-	renderer, err := resolve.NewPlainVariableRendererWithValidationFromTypeRef(p.visitor.Operation, p.visitor.Operation, variableTypeRef, string(variableName))
-	if err != nil {
-		return
-	}
-	contextVariable := &resolve.ContextVariable{
-		Path:     []string{string(variableName)},
-		Renderer: renderer,
-	}
-	variablePlaceHolder, exists := p.variables.AddVariable(contextVariable) // $$0$$
-	if exists {
-		return
-	}
-	// We need to replace the template literal with the variable placeholder
-	p.current.topic = rex.ReplaceAllLiteralString(topic, variablePlaceHolder)
-
-	// Collect the field arguments for fetch based operations
-	fieldArgs := p.visitor.Operation.FieldArguments(ref)
-	var dataBuffer bytes.Buffer
-	dataBuffer.WriteByte('{')
-	for ii, arg := range fieldArgs {
-		if ii > 0 {
-			dataBuffer.WriteByte(',')
-		}
-		argValue := p.visitor.Operation.ArgumentValue(arg)
-		renderer := resolve.NewJSONVariableRenderer()
-		variableName := p.visitor.Operation.VariableValueNameBytes(argValue.Ref)
-		contextVariable := &resolve.ContextVariable{
-			Path:     []string{string(variableName)},
-			Renderer: renderer,
-		}
-		variablePlaceHolder, _ := p.variables.AddVariable(contextVariable)
-		argumentName := p.visitor.Operation.ArgumentNameString(arg)
-		escapedKey, err := json.Marshal(argumentName)
-		if err != nil {
-			return
-		}
-		dataBuffer.Write(escapedKey)
-		dataBuffer.WriteByte(':')
-		dataBuffer.WriteString(variablePlaceHolder)
+	switch eventConfig.Type {
+	case EventTypePublish:
+		fallthrough
+	case EventTypeRequest:
+		p.handlePublishAndRequestEvent(ref, eventConfig)
```

Comment:
You can make it more explicit like this I think:

```go
case EventTypeRequest, EventTypePublish
```

Reply:
Done